django/contrib/admin/options.py

```diff
@@ -268,10 +270,14 @@ def get_view_on_site_url(self, obj=None):
             return self.view_on_site(obj)
         elif self.view_on_site and hasattr(obj, 'get_absolute_url'):
             # use the ContentType lookup if view_on_site is True
-            return reverse('admin:view_on_site', kwargs={
+            current_lang = get_language()
+            activate(obj.language_code)
```

Comment:
You can use the `with translation.override(obj.language_code)` context manager to simplify it.